### PR TITLE
enh: export only selected resources

### PR DESCRIPTION
**Pull request by sc979**
Export only the selected resources in the CSV file
Add a warning when less than 1 resource is selected
Add checks and sanitize different parameters

19.10.x
20.04.x

fixes: (MON-5010)

### CODE REVIEW

File: host-monitoring/data.js [hunk 1]
```diff
@@ -72,3 +72,27 @@ jQuery(function () {
...
 });
+
+/**
+ * retrieve selected resources
+ */
+function exportChecked() {
+    let exportList = '';
+    // get checked resource list from local storage
+    $(".selection").each(function () {
+        var itemSaved = 'w_hm_' + $(this).attr('id');
```
Comment (sc979) [inline host-monitoring/data.js:83]:
```suggestion
        let itemSaved = 'w_hm_' + $(this).attr('id');
```

File: host-monitoring/src/export.php [hunk 2]
```diff
@@ -69,15 +71,38 @@
 $template = initSmartyTplForPopup($path, $template, './', $centreon_path);
 
 $centreon = $_SESSION['centreon'];
-$widgetId = filter_input(INPUT_REQUEST, 'widgetId', FILTER_VALIDATE_INT, ['options' => ['default' => 0]]);
+$widgetId = filter_input(INPUT_GET, 'widgetId', FILTER_VALIDATE_INT, ['options' => ['default' => 0]]);
+
+// Sanitize and concatenate selected resources for the query
+if (false !== strpos($_GET['list'], ',')) {
```
Comment (callapa) [inline host-monitoring/src/export.php:77]: No filter for $_GET['list'] ?

File: host-monitoring/src/export.php [hunk 3]
```diff
@@ -69,15 +71,38 @@
...
 $mainQueryParameters = [];
+$hostQuery = '';
+foreach ($exportList as $key => $hostId) {
+    if (0 === (int) $hostId) {
+        continue;
+    }
+    if (!empty($hostQuery)) {
+        $hostQuery .= ', ';
+    }
+    $hostQuery .= ':' . $key . $hostId;
```
Comment (callapa) [inline host-monitoring/src/export.php:91]: What's happen if:
key = 1 & hostId = 11 => hostQuery = 111
key =11 & hostId = 1 => hostQuery = 111

File: host-monitoring/src/export.php (hunk 2 above)
Comment (sc979) [reply host-monitoring/src/export.php:77]: not here. It's done later in this block: https://github.com/centreon/centreon-widget-host-monitoring/pull/87/files#diff-dfd0cade38704c4b43b925be7a8f5935R85
This part is intended to prepare an array when only one value is returned in the params

File: host-monitoring/src/export.php (hunk 3 above)
Comment (sc979) [reply host-monitoring/src/export.php:91]: you're right

File: host-monitoring/src/export.php [hunk 4]
```diff
@@ -69,15 +71,38 @@
 $template = initSmartyTplForPopup($path, $template, './', $centreon_path);
 
 $centreon = $_SESSION['centreon'];
-$widgetId = filter_input(INPUT_REQUEST, 'widgetId', FILTER_VALIDATE_INT, ['options' => ['default' => 0]]);
+$widgetId = filter_input(INPUT_GET, 'widgetId', FILTER_VALIDATE_INT, ['options' => ['default' => 0]]);
+
+// Sanitize and concatenate selected resources for the query
```
Comment (sc979) [inline host-monitoring/src/export.php:76]:
```suggestion
/**
 * Sanitize and concatenate selected resources for the query
 */
 // Check returned list and make an array of it
```

File: host-monitoring/src/export.php [hunk 5]
```diff
@@ -69,15 +71,41 @@
...
+ */
+ // Check returned list and make an array of it
+if (false !== strpos($_GET['list'], ',')) {
+    $exportList = explode(',', $_GET['list']);
+} else {
+    $exportList[] = $_GET['list'];
+}
 $mainQueryParameters = [];
+$hostQuery = '';
+foreach ($exportList as $key => $hostId) {
```
Comment (sc979) [inline host-monitoring/src/export.php:87]:
```suggestion
// Check consistency, sanitize and bind values
foreach ($exportList as $key => $hostId) {
```

File: host-monitoring/src/export.php [hunk 6]
```diff
@@ -69,15 +71,42 @@
...
+    $exportList = explode(',', $_GET['list']);
+} else {
+    $exportList[] = $_GET['list'];
+}
 $mainQueryParameters = [];
+$hostQuery = '';
+// Check consistency, sanitize and bind values
+foreach ($exportList as $key => $hostId) {
+    if (0 === (int) $hostId) {
+        continue;
```
Comment (sc979) [inline host-monitoring/src/export.php:90]:
```suggestion
        // skip non consistent dat
        continue;
```

File: host-monitoring/src/export.php (hunk 2 above)
Comment (callapa) [reply host-monitoring/src/export.php:80]: Ok but with `$exportList[] = $_GET['list'];`
What's happen if:
`$_GET['list'] => blablabla,1` here the key is not numerical and later you did `$hostQuery .= ':' . $key . 'host' . $hostId;` without filtering the key

**Pull request merged**